unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,348 @@
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata=additional_metadata,
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    def is_async(self) -> bool:
+        return True
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path: str):
+        """Fetches a file from the GitHub repository using the GitHub API.
+
+        Args:
+            path (str): The path to the file in the repository.
+
+        Returns:
+            content_file: The file content object.
+
+        Raises:
+            UnknownObjectException: If the file does not exist in the repository.
+        """
+        from github.GithubException import UnknownObjectException
+
+        try:
+            logger.info(f"Fetching file from path: {path!r}")
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exist: {self.connection_config.url}/{path}")
+            raise UnknownObjectException
+
+        return content_file
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["httpx", "github"], extras="github")
+    async def _fetch_content(self, content_file):
+        """Asynchronously retrieves the content of a file, handling large files via direct download.
+
+        Args:
+            content_file: The file object from the GitHub API.
+
+        Returns:
+            bytes: The content of the file as bytes.
+
+        Raises:
+            UnknownObjectException: If the download or content retrieval fails.
+        """
+        import httpx
+        from github.GithubException import UnknownObjectException
+
+        contents = b""
+        async with httpx.AsyncClient() as client:
+            if not content_file.content and content_file.encoding == "none" and content_file.size:
+                logger.info(
+                    "File too large for the GitHub API, using direct download link instead."
+                )
+                try:
+                    response = await client.get(content_file.download_url, timeout=10.0)
+                    response.raise_for_status()
+                except (httpx.HTTPStatusError, httpx.RequestError) as e:
+                    logger.error(f"Failed to download {content_file.download_url}: {e}")
+                    raise UnknownObjectException
+
+                contents = response.content
+            else:
+                contents = content_file.decoded_content
+        return contents
+
+    async def _fetch_and_write(self, path: str, download_path: str) -> None:
```

Comment:
Since the download path when you call this method passes in a pathlib.Path object rather than a string. 

```suggestion
    async def _fetch_and_write(self, path: str, download_path: Path) -> None:
```